packages/core/useMediaQuery/index.ts

```diff
@@ -23,14 +24,12 @@ export function useMediaQuery(query: string, options: ConfigurableWindow = {}) {
   const update = () => {
     if (!isSupported.value)
       return
-    if (!mediaQuery)
-      mediaQuery = window!.matchMedia(query)
+    mediaQuery = window!.matchMedia(resolveRef(query).value)
     matches.value = mediaQuery.matches
   }
+  watchEffect(update)
 
   tryOnBeforeMount(() => {
-    update()
-
     if (!mediaQuery)
       return
 
```

Comment:
It also should be reactive https://github.com/vueuse/vueuse/pull/2178/files#diff-d9533c610c4da5a1c5a0e03fc40091a2a8834c552c81ffc974287a7d4f7d53f1L37-L49